docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer.md

```diff
@@ -0,0 +1,75 @@
+| AOT support                                    | ❌️              | ✔️                      | ❌                    | ✔️                       | ❌ (planned)                     |
+
+## JSON using System.Text.Json
+
+The [`System.Text.Json`](../system-text-json/overview.md) library is a modern serializer that emphasizes security, high performance, and low memory allocation for the JavaScript Object Notation (JSON) format. JSON is human-readable and has broad cross-platform support. While text-based format is not as compact as binary formats, it can be significantly reduced in size through compression.
+
+Serialization excludes non-public and readonly members unless specifically handled through attributes and constructors. System.Text.Json also supports [custom serialization and deserialization](../system-text-json/custom-contracts.md) for more control over how types are converted into JSON and vice versa. System.Text.Json does not support the `[Serializable]` attribute.
+
+[Migrate to System.Text.Json](./migrate-to-system-text-json.md).
+
+## XML using System.Runtime.Serialization.DataContractSerializer
+
+`DataContractSerializer` was introduced in .NET Framework 3.0 and is used to serialize and deserialize data sent in Windows Communication Foundation (WCF) messages. `DataContractSerializer` is an XML serializer that **fully supports the serialization programming model that was used by the `BinaryFormatter`**. It requires the known types to be specified up-front (but most .NET collections and primitive types are on a default allow-list and don't need to be specified). It's the serializer that requires the least amount of effort to migrate to.
+
+While `DataContractSerializer` carries those functional benefits when migrating from BinaryFormatter, it is not as performant as the other choices, nor does it use a cross-platform format.
+
+[Migrate to DataContractSerializer](./migrate-to-datacontractserializer.md).
+
+> [!NOTE]
+> Do not confuse `DataContractSerializer` with [`NetDataContractSerializer`](/dotnet/api/system.runtime.serialization.netdatacontractserializer). `NetDataContractSerializer` is also identified as a [dangerous serializer](../binaryformatter-security-guide#dangerous-alternatives).
+
+## Binary using MessagePack for C#
+
+MessagePack uses a highly efficient binary serialization format, resulting in smaller message sizes compared to JSON and XML. It's performant and ships with built-in LZ4 compression. The open-source [MessagePack for C#](https://github.com/MessagePack-CSharp/MessagePack-CSharp) library works best when all serializable public types and members are annotated with dedicated attributes. It does not serialize non-public types and members by default, but it can be customized. It supports read-only types and members by trying to select the best matching constructor. The constructor can be specified explicitly by using the `[SerializationConstructor]` attribute.
+
+[Migrate to MessagePack](./migrate-to-messagepack.md).
```

Comment:
@AArnott Can you review this portion please? Feel free to push refinements directly to the PR/branch as you see fit.